### PR TITLE
Reports enhanced to support SNOMED Feature

### PR DESCRIPTION
**Pull request by vijayanandtwks**
## Requirements

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more readable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or atleast performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Summary
Configuration/Generation of generic and gender based SNOMED Reports for diagnosis count

## Screenshots
image

## JIRA tickets
https://bahmni.atlassian.net/browse/BS-24
https://bahmni.atlassian.net/browse/BS-51
https://bahmni.atlassian.net/browse/BS-55

### CODE REVIEW

File: openmrs/apps/reports/reports.json [hunk 1]
```diff
@@ -115,7 +115,7 @@
             ],
             "showVisitInfo": true,
             "showProvider": true,
-            "formNamesToFilter": [ "Test Form"],
+            "formNamesToFilter": [ "Test Form", "SNOMED Form", "SNOMED Obs Form"],
```
Comment (gsluthra) [inline openmrs/apps/reports/reports.json:118]: Can we give a more realistic name. What FORM is this? H&E, Pain Form, Malaria form, etc.  If this is a new H&E form, we can give it a name like: "H&E Form (SNOMED)", so it is clear this is a History and Examination Form, but supports SNOMED fields.

File: openmrs/apps/reports/reports.json [hunk 2]
```diff
@@ -139,6 +139,39 @@
             ]
         }
     },
+    "diagnosisCountMalaria": {
+        "name": "SNOMED Diagnosis Count Report for Malaria",
+        "type": "tsIntDiagnosis",
```
Comment (gsluthra) [inline openmrs/apps/reports/reports.json:144]: I recommend the name should be: `fhirTSDiagnosis`, or `fhirTSLookupDiagnosis`.

File: openmrs/apps/reports/reports.json [hunk 3]
```diff
@@ -139,6 +139,39 @@
...
+        }
+    },
+    "diagnosisCountAsthma": {
+        "name": "SNOMED Diagnosis Count Report for Asthma",
+        "type": "tsIntDiagnosis",
+        "requiredPrivilege": "app:reports",
+        "config": {
+            "tsConceptSource": "SNOMED",
+            "terminologyParentCode": "195967001",
+            "displayTerminologyCode": "true",
```
Comment (gsluthra) [inline openmrs/apps/reports/reports.json:160]: Should have an attribute called: `"terminologyColumnName": "SNOMED Code"`

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: openmrs/apps/reports/reports.json (hunk 1 above)
Comment (vijayanandtwks) [reply openmrs/apps/reports/reports.json]: Based on our milestone demo inputs, i have made the form as follows
[ "Test Form", "SNOMED H&E Form", "SNOMED Malaria Form", "SNOMED Asthma Form", "SNOMED COVID Form", "SNOMED Accident Intake Form"]

**Review by gsluthra — APPROVED** [review]

**Pull request merged**